### PR TITLE
[PropertyInfo] Add non-*-int missing types for PhpStanExtractor

### PR DESCRIPTION
**Pull request by wuchen90**
| Q             | A
| ------------- | ---
| Branch?       | 7.3
| Bug fix?      | no
| New feature?  | yes
| Deprecations? | no
| License       | MIT

This PR adds support for following PHPStan types in Property Info component:
- `non-positive-int`
- `non-negative-int`
- `non-zero-int`

See PHPStan official docs for these types: https://phpstan.org/writing-php-code/phpdoc-types#integer-ranges

### CODE REVIEW

**Review by mtarld — APPROVED** [review]

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
I'm going to merge it in 7.3 as this is a new feature.

**Comment by nicolas-grekas** [comment]
@wuchen90 to match @fabpot's comment, could you please rebase on 7.3 and add a line in the changelog of the component in the 7.3 section? (to be created potentially)

**Comment by derrabus** [comment]
Also, please rebase onto 7.3 and resolve the conflicts.

**Review by stof — APPROVED** [review]

File: src/Symfony/Component/PropertyInfo/CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,11 @@
 CHANGELOG
 =========
 
+7.3
+---
+
+* Add `non-positive-int`, `non-negative-int` and `non-zero-int` PHPStan types for `PhpStanExtractor`
```
Comment (xabbuh) [inline src/Symfony/Component/PropertyInfo/CHANGELOG.md]:
```suggestion
 * Add support for `non-positive-int`, `non-negative-int` and `non-zero-int` PHPStan types to `PhpStanExtractor`
```

**Review by xabbuh — APPROVED** [review]
with a minor CS change in the changelog

**Review by derrabus — APPROVED** [review]

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @wuchen90.

**Pull request merged**